packages/loader/container-loader/src/container.ts

```diff
@@ -674,17 +675,65 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         this.resumeInternal();
     }
 
+    /**
+     * Waits until container connects to delta storage and gets up-to-date
+     * Useful when resolving URIs and hitting 404, due to container being loaded from (stale) snapshot and not being
+     * up to date. Host may chose to wait in such case and retry resolving URI.
+     * @returns true: container is up to date, it processed all the ops that were know at the time of first connection
+     *          false: storage does not provide indication of how far the client is.
+     */
+    public async waitCatchUp() {
```

Comment:
>waitCatchUp [](start = 17, length = 11)

is is possible to make this a utility function, rather than part of container? Allowing this to built on the public surface of the container would allow others to build their own with custom policies, like retry

Reply:
saw your comment below. still a good idea

---
In reply to: [500451980](https://github.com/microsoft/FluidFramework/pull/3830#discussion_r500451980) [](ancestors = 500451980)

Reply:
Moved it out into stand-alone function